### DISCUSSION TITLE
Use no-go areas in route planning

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#538] [Q&A] [answered]
Route planning with **no-go areas** arrives in Cruiser:

- Add no-go areas (long press on map)

- No-go areas have their own menu:

  - Move them (precisely with a new crosshair)

  - Resize them (change their radius)

  - Delete them

- No-go areas are saved in favorites: https://github.com/devemux86/cruiser/discussions/545

No-go areas work similarly to the BRouter website.

An example of routing with no-go areas:

Before|After
-|-
<img src="https://github.com/user-attachments/assets/695c68ac-f925-47bd-8c44-f664cced8d6a"/>|<img src="https://github.com/user-attachments/assets/904cd6a3-7025-4356-978b-11a853a12a40"/>
<img src="https://github.com/devemux86/cruiser/assets/3484020/1a53b92e-e4a2-4ab7-810b-1ee24eb1c844"/>|<img src="https://github.com/devemux86/cruiser/assets/3484020/a5554eab-228d-4bf3-ab61-330ffa590fbf"/>

**Comment by devemux86** [comment]
(open external BRouter no-go files will be removed)

**Comment by devemux86** [comment]
> If you could add a feature for NoGoAerias, similar to Favorites?
Select a NoGoAeria with crosshair, add radius,name.
Saved in “my collection”.

_Originally posted by @Leeb64._

**Reply by devemux86** [reply]
Why would we need to store the no-go areas in favorites with names?
This requires a lot of work to implement UI and underlying structure.

No-go areas are essential part of the routes, like waypoints.
So they can be saved / loaded / transferred inside the routes.

**Comment by WolfgangZ42** [comment]
Okay - this are no-go-areas for planning a new route. **Question:** will these no-go-area definition stored as part of the route - for later changes? Sorry - is already answered in the first post.
And I'm using a polygon no-go-area since more as a year - see: https://github.com/devemux86/cruiser/discussions/85#discussioncomment-5577744 and it works! Today I can already select stored no-go-areas by define them with _/Routing/Options/Advanced_:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/d06f3d32-43bc-4680-b0a2-74e6c5d5f4b7"/>

**Reply by devemux86** [reply]
> will these no-go-area definition stored as part of the route - for later changes?

No-go areas are part of the routes and handled like the waypoints,
so they are saved / loaded / transferred like them with the routes.

> I'm using a polygon no-go-area

This setting will eventually be removed, as it complicates the process. :slightly_smiling_face: 
It is also not possible to draw arbitrary shapes, unless someone provides a ready no-go file.

Drawing circular no-go areas like BRouter website should be able to cover all cases.

**Reply by WolfgangZ42** [reply]
> unless someone provides a ready no-go file.

I've a no-go file and I used it in my car profile - now it's gone ... 😥

Why couldn't kept this kind of defining a no-go-area for a profile (and all routes using this profile) additionally to the route-related no-go-area? I think Cruiser is the only navigation app that offered such a feature - a permanently defined no-go-area! 
Sure the route-related no-go-area is important as well. And it's perfectly implemented, no question. But only stored in navigatable tracks - right?
Please think about providing two different no-go-area definitions ...

**Reply by devemux86** [reply]
@WolfgangZ42 it is complicated to mix such different implementations.
(that was just a temporary solution until no-go areas are implemented properly)

We can think about how to optimally use the no-go areas in Cruiser:

- No-go areas are now saved / loaded / transferred with the routes.

- @Leeb64 mentioned storing them in favorites, although it is much more work.

- We could also separate the route from the no-go areas in the clear dialog:
(like in BRouter website)

![BRouter](https://github.com/devemux86/cruiser/assets/3484020/bb2c72cd-a5ea-4813-b082-f6eebb76ad54)

**Reply by WolfgangZ42** [reply]
Where does BRouter store the no-go areas? And is it kept forever? Can't believe it. Stopping the app, restarting Android, etc. 

Up to now, no-go-areas are .gpx-files for me. Either a circular defined wpt or a polygon:
![image](https://github.com/devemux86/cruiser/assets/121806238/b267301d-8a17-4e3b-875e-cd7d09bff2b4)

Here two .gpx-files of these kinds: [nogo5000_Munich.gpx.txt](https://github.com/devemux86/cruiser/files/15405871/nogo5000_Munich.gpx.txt)   [nogo_Munich_MR.gpx.txt](https://github.com/devemux86/cruiser/files/15405878/nogo_Munich_MR.gpx.txt)

I've stored these files on my SDcard and live with it since more than a year. And the best of this "temporary solution" was the ability to use it only in my car profile.

**Reply by devemux86** [reply]
@WolfgangZ42

Android adds more and more restrictions in new versions, we better avoid complicated things with external files.

If there is need to store and use easily the no-go areas, we could implement it inside Cruiser, e.g. in favorites.

Even advanced users cannot edit text files trying to define borders of a city, this is not a simple way to work.
We should promote the app interface so all users can follow and use it easily. :slightly_smiling_face:

**Reply by devemux86** [reply]
Let's not mix Android with desktop, as users expect everything from the mobile app.

Also it is not easy to design complicated shapes on Android just with hand gestures.
Circles need only center and radius, so they are simple and can cover all no-go cases.

The need to store and reuse them seems similar to the favorites used as waypoints.

**Reply by WolfgangZ42** [reply]
Not insisting in the polygon no-go-area (btw: I've a tablet with keyboard and mouse ...) but would like to define no-go-areas permanently for a specific profile and not just for a route. Again: consider road-work for several months or avoiding a downtown area of a city. 
Nevertheless the route-related no-go-areas are absolutely usefull as well.

**Reply by devemux86** [reply]
Favorites are global for all profiles, so it not simple to change now their structure and separate them.

If we have favorite no-go areas, we could use them easily from favorites?
Just like we use the favorite points to set waypoints in route planning.

---

About the favorite no-go areas, I created a new topic as it is a different feature,
please let's discuss them there:

- https://github.com/devemux86/cruiser/discussions/545

**Reply by 0709wiwiwi** [reply]
Non route related fixed nogo areas.

> not easy to design complicated shapes on Android just with hand gestures.

To design a polygon nogo rte or trk by a mobile unit is basic simple.
Tap routepoint 1 and 2 by straight line and continue by extend route. 
**In the current beta, the extend route button is only accessible after 2 taps.
That the design buttons are hidden deeper becomes quite annoying.
The unreserved 3x4 button panel offered more opportunities for this.**

**Reply by devemux86** [reply]
3x4 panels can work nicely when there are no texts in the buttons, but only icons.

**Reply by 0709wiwiwi** [reply]
I also followed the all-too-long buttons and panel discussion. 
It was immediately clear that you cannot fix this definitively.
The deeper buttons are hidden the more difficult it becomes.

**Comment by devemux86** [comment]
@WolfgangZ42

Please see [Cruiser 4.1.0-b2](https://github.com/devemux86/cruiser/discussions/544#discussioncomment-9530401) where BRouter external no-go files can work with internal no-go areas.

**Comment by bios4** [comment]
> * We could also separate the route from the no-go areas in the clear dialog

👍 from my side for that suggestion - it would be great to have the option to KEEP set up no-go areas INTACT when deleting a route (via Main menu > Routing > Clear):

<img src="https://github.com/devemux86/cruiser/assets/145460147/5967deaf-bdcb-4b8e-9e3a-8342f99003ce" width="33%">

**Reply by devemux86** [reply]
@bios4 it is indeed an interesting suggestion.

The workflow right now is that the no-go areas are saved with the route data.
So it needs changes in the structure to work differently and exist on their own.

e.g. we set some waypoints and the route calculation fails for whatever reason.
When we restart the app the waypoints have not been saved. Similar case here.

**Reply by bios4** [reply]
> when deleting a route (via Main menu > Routing > Clear)

And also after long-pressing the Route > Clear... 😉

**Reply by bios4** [reply]
Regarding the German translation of the new function: simply display "Verbotene Bereiche" (instead of "Farbe des verbotenen Bereichs") in Main > Settings > Routing > Appearance > Colors...

**Comment by WolfgangZ42** [comment]
> the option to KEEP set up no-go areas INTACT when deleting a route

Not only that - why not save them for later _use?_

**_Status quo:_**
The no-go areas defined within the route planning in circle format are automatically displayed and saved with the navigable track (but also in JSON and Kurviger format). In addition, there are also the "permanent" no-go areas (circles and polygons) which are located inform of gpx files in a directory and can be activated via _/Routing/Options/Advanced/No-go areas_.

**_Suggestion:_**
Now I have learned (thanks Willy) that it is also possible to define tracks as no-go areas. I promptly changed my polygon for the Munich Diesel ban zone in a track version. Advantage: I can display the no-go zone (as well as all other no-go tracks) via _/My collection/Tracks_ and a separate group. See my example:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/5fa45878-831a-4976-bae4-fa6a0247ba5f"/>

A separate category in _/My collection_ called _No-go areas_ for tracks and circles would be perfect:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/9cbad76e-c2df-4ce9-bf3e-a90c7210c48c"/>
 
No-go tracks could be inserted like into _/My collection/Tracks_. Circles could be defined as done now in the planning process and additionally saved permanently, whereby a name must of course be assigned (e.g. "nogo800 Downtown Augsburg"). Something like this:
<img width="200" src="https://github.com/devemux86/cruiser/assets/121806238/bc430405-2c1b-47c8-9938-e0e8c058da13"/>
Then we can use this no-go area not only in the current route but also in upcoming routes in the future. 

Activate all relevant no-go areas from this category profile-dependent via _/Routing/Options/Advanced/No-go areas_ and then also display it on the map. 

No high priority, I'm one of few users who deal with no-go areas and can live with the current situation as well.

**Reply by devemux86** [reply]
> define tracks as no-go areas

Note that they are not the same:
- No-go areas block routing inside their bounds
- No-go tracks block routing only on their lines (not within their perimeter)

**Reply by devemux86** [reply]
@WolfgangZ42 thanks, you described the needs and possible functions very well! :+1:

**Comment by umpfmpf** [comment]
Is it also possible to define individual points from a gpx file as a no-go area? In my case, however, this file may change daily and should not be saved permanently in the favorites? It should only be used for this one day and on the next day should a new file be used.
[poi_export.gpx.txt](https://github.com/user-attachments/files/16044826/poi_export.gpx.txt)

**Reply by devemux86** [reply]
We will see about the format content, as there is no standard.

Anyway the app has to remember the no-go areas, so they need to be saved, could be in favorites.
Then they could be enabled / disabled or deleted again.

**Reply by devemux86** [reply]
No-go areas must have location + size.
The simplest form is a circle with center + radius, like in BRouter website.

A gpx with simple waypoints does not describe their size, just their location.
Unless a radius of default X meters is used.

**Comment by devemux86** [comment]
Next Cruiser can handle all types of shapes as no-go areas, like points / circles and polygons.

I updated the screenshots in the [announcement](https://github.com/devemux86/cruiser/discussions/538#discussion-6671292), please check them!

As you can see, the route planning in Cruiser now becomes extremely detailed and intelligent. :slightly_smiling_face: 

---

All no-go area types can be stored in "My collection" and turn on / off their no-go function:
- https://github.com/devemux86/cruiser/discussions/545

**Comment by devemux86** [comment]
No-go areas as polygons are supported in addition to BRouter and in Valhalla and OpenRouteService routing.

**Comment by devemux86** [comment]
Use no-go areas in route planning is available in:

- https://github.com/devemux86/cruiser/discussions/610

- Instructions: https://github.com/devemux86/cruiser/discussions/538#discussion-6671292

**Comment by Ciutnik** [comment]
This works in Cruiser 5. Thanks! 👍
Unfortunately, it works globally, in all profiles?

And a second thought: I only have a problem now with several "nogo-tracks" that I have defined for myself on roads that I never want to drive. In Cruiser 4 I loaded them together with areas (which were also de facto tracks). Now I can replace them with a "nogo-circle", but this may not always be a good solution.

**Reply by devemux86** [reply]
> Unfortunately, it works globally, in all profiles?

@Ciutnik What do you mean?

Only the visible no-go areas (circles or polygons) participate in routing.
Now you can change their visibility per app profile:
- https://github.com/devemux86/cruiser/discussions/597

> nogo-tracks

BRouter also supports no-go tracks, I just have not enabled it yet. Do we need it?

Note that setting too many and complex no-go areas make routing much slower.
Polygons and tracks can be very complex compared to the simpler no-go circles.

**Reply by Ciutnik** [reply]
> Now you can change their visibility per app profile

Actually, you're right, I'm sorry, but somehow I didn't notice at first...! I think I got confused by the "nogo" symbol I see next to the area. But turning off visibility also disables the use of this area as a "nogo". Okay, now that I know, it's no problem and I like it 😁

> BRouter also supports no-go tracks, I just have not enabled it yet. Do we need it?

For now, I have replaced nogo-tracks with small nogo-circles. That's enough for now. It's probably just a matter of visibility on the map. It's easier to spot a long track on the road than a tiny circle.

Most importantly, avoidance can be easily achieved in routing and navigation.Overall, the ability to define unique areas is extremely useful and well implemented! Thanks a lot 🙂

**Reply by devemux86** [reply]
> Now you can change their visibility per app profile

@Ciutnik currently there is a [reported problem](https://github.com/devemux86/cruiser/discussions/610#discussioncomment-10243916) we first need to fix.

> It's probably just a matter of visibility on the map. It's easier to spot a long track on the road than a tiny circle.

We could probably display something like a symbol in small zooms to indicate the presence of no-go circles.

**Reply by Ciutnik** [reply]
@devemux86 
> We could probably display something like a symbol in small zooms to indicate the presence of no-go circles.

You can do it on your own. Here is a combination of a 50 meter circle, track (until now it was nogo, now only visible) and an icon 🙂 

<img src="https://github.com/user-attachments/assets/a01e66d1-e5f6-433a-a8c6-319cb5b46b69" width="75%"/>

**Reply by DIrgi** [reply]
Don't get me wrong but where do you guys have all this "no-go" zones?
If they are existing already in the maps (i.e. military zones, private areas), they are most likely not open for public routing so they won't work anyway and if we talk about blocked tracks/roads for what we normally do - These are no zones as well but blocked roads and tracks like here in the picture the dotted line (only). 

I rather see zones when it comes to environment reasons but then they most likely depend on the vehicle you drive and they are most likely not a circle but following borders of an (city, county etc.)area.

**Reply by devemux86** [reply]
@DIrgi it's not only for defined "zones" by organizations.

No-go areas (as demonstrated also in [BRouter website](https://brouter.de/brouter-web/)) are extremely useful to freely customize the routing.
- We use shaping points to make routes _going where we want_

- We use no-go areas to make routes _**NOT** going where we want_
e.g. avoid cities, villages, already visited areas or roads, ... The possibilities are endless.

**Reply by DIrgi** [reply]
I see but then it's just like defining Start and End point and let the system then calculate by avoiding these "areas".
Ok, I never plan like this I use at least shaping points to force the system where I want to go or not and exclude the "no go zones" then by myself in advance but if you mainly let the system decide by creating a random route, this might make sense.

**Comment by umpfmpf** [comment]
I imported my blocked motorcycle routes into the no-go areas and they are now displayed and bypassed properly. However, only the 2 coordinates that are specified, i.e. start and end point, are displayed as points, but it works.

**Reply by DIrgi** [reply]
I have them as well, blocked roads (most likely depending on the day/time per year and vehicle) but I have them only displayed as information and not blocked, because they are quite often not blocked 24/7.

Example:
The red line shows you the blocked section and clicking on the symbol it display short information when the road is blocked (all for motorcycles) and only on saturday/sunday. During the week it's free. So routing during the weekend if I define it as no go road/zone it will as well guide me around. 

I assume for the "no go areas" there cannot used any time/day period which should be in consideration - blocked is bocked - or?

![image](https://github.com/user-attachments/assets/39af6ac5-0513-41ef-ab57-b3fb84ecf3f9)

**Reply by umpfmpf** [reply]
Looks good. I normally only have the blocked routes as pois, but only the start and end points. How did you fill the routes in between?

**Reply by DIrgi** [reply]
Gimme A second I can share

**Reply by devemux86** [reply]
> I assume for the "no go areas" there cannot used any time/day period which should be in consideration

Probably, but isn't it easier to have them in a group and switch group's visibility on / off?

**Reply by DIrgi** [reply]
unfortunately not really - depending on the no-go area/ road. 
If it'll be a road/zone what you always like to block all the time forever then yes, but if a block should be timewise and your routing should have it in consideration, a lot of road blockings are different some only on several days some completely and some only during a (summer) period or only from 7am to 7pm and so on.

Even the kurviger desktop page just show you the information that a road might be blocked but allows you to plan because kurviger never knows for which vehicle you plan and which time.

A couple of years ago I used google maps for route planning then I recognized when I plan in winter a tour what I would like to drive in the summertime via some passes in the alps, Google always guided me around, because in winter time (so present, right now while I'm planning) the road was blocked. I had google to tell that I plan for summer time (by setting the departure time to summer) then it works but after all it was to complicated an google maps is not really a planning tool for a long time route for several days.

**Reply by devemux86** [reply]
We already have the departure date / time as a feature idea,
so it could be combined if we enrich the no-go areas with a date / time period.

- https://github.com/devemux86/cruiser/discussions/606

**Reply by DIrgi** [reply]
yes, but ones again. Finally it'll be a hell of work due to many different regulations. I assume nobody will really use this function or at least just for a few of them. 
At the end to be precise you have to define for every blocked area route/area it's own boundaries and then it's 100% perfect and routing will have it in consideration in that way that maybe today you can drive to road and tomorrow only during 7am to 6pm


Example (extraction of road closures)

```
<wpt lat="52.1570778" lon="9.0322208">
    <time>2011-09-01T16:39:34Z</time>
    <name>31737 SaSo -Eiergrund</name>
    <cmt>1.4. - 30.9.</cmt>
    <desc>1.4. - 30.9.</desc>
    <sym>Navaid, Green/Red</sym>
    <extensions>
      <gpxx:WaypointExtension xmlns:gpxx="http://www.garmin.com/xmlschemas/GpxExtensions/v3">
        <gpxx:DisplayMode>SymbolOnly</gpxx:DisplayMode>
      </gpxx:WaypointExtension>
    </extensions>
  </wpt>
  <wpt lat="52.1497123" lon="9.0679531">
    <time>2011-09-18T19:32:30Z</time>
    <name>31737 SaSo -Möllenbeck</name>
    <cmt>1.4. - 30.9.</cmt>
    <desc>1.4. - 30.9.</desc>
    <sym>Navaid, Green/Red</sym>
    <extensions>
      <gpxx:WaypointExtension xmlns:gpxx="http://www.garmin.com/xmlschemas/GpxExtensions/v3">
        <gpxx:DisplayMode>SymbolOnly</gpxx:DisplayMode>
      </gpxx:WaypointExtension>
    </extensions>
  </wpt>
  <wpt lat="52.2798110" lon="9.4929425">
    <time>2011-09-18T19:32:41Z</time>
    <name>31848 So -Nienstedt</name>
    <cmt>6 - 22 Uhr</cmt>
    <desc>6 - 22 Uhr</desc>
    <sym>Navaid, Green/Red</sym>
    <extensions>
      <gpxx:WaypointExtension xmlns:gpxx="http://www.garmin.com/xmlschemas/GpxExtensions/v3">
        <gpxx:DisplayMode>SymbolOnly</gpxx:DisplayMode>
      </gpxx:WaypointExtension>
    </extensions>
  </wpt>
```

So, the first and second ones are blocked on ALL Saturdays and Sundays BUT ONLY between 1.4. and 30.9
The third one is blocked on Sunday only and only between 6 and 22h.

These are just some examples....

**Comment by devemux86** [comment]
We could also allow the lines to be used as no-go areas (BRouter seems to support it).
Not sure about the performance as lines are always complex geometries.

Now we have no-go circles and polygons.

**Reply by umpfmpf** [reply]
That wouldn't be bad, as some routes also have intersections and you're still not allowed to drive there. That would be better than just marking the end points.

**Comment by DIrgi** [comment]
@umpfmpf : You have to fulfill two criteria's.

1. You have to have a POI to mark the information (what you already have) and
2. You have to have an additional track to mark the section which is blocked.

Both together have to be displayed then it works. One is the informal part and one the visible one.


Here are the files:
Unzip

[DACH temp road blokings.zip](https://github.com/user-attachments/files/16521355/DACH.temp.road.blokings.zip)

put the POI file into your Favorites folder, make it visible, define your zoom level, give a red color, select your own sign.
put the TRACKS file into your track/area folder, generate maybe a sub folder "road closure" put the file there and do the same like you did with the POIs - et voila now you have all available road closures for DACH - file status is from 04/2024

have fun

**Comment by umpfmpf** [comment]
Is it possible to create multiple ornaments with no-go areas?One for cities, one for weekend closures, one for complete closures and one for seasonal closures, such as in the mountains?
Thanks to @DIrgi

**Reply by DIrgi** [reply]
For sure it is, several folders for several belonging and each folder has to tell the BRouter what to do and when to allow/block the containing data.

Here we are again - Who will do this or better is everybody willing to maintain (just) his own data in that way. Don't get me wrong, the idea is closer to an reality but I wonder if everybody or many will use it in that way.

**Reply by DIrgi** [reply]
Just to remember - Even the "big players" don't offer these features to customers (at least it's not known to me). They only inform them (that a road might be closed), or they use real live data for road closures but I assume even Google won't consider that roads are blocked for motorbikes only and only between November to May and then only between 8am and 6pm. If they do it, they do it in a more global scope

**Reply by umpfmpf** [reply]
> For sure it is, several folders for several belonging and each folder has to tell the BRouter what to do and when to allow/block the containing data.
> 

That was my question: is this possible in Cruiser 5 with certain folder names (Forbidden Areas, ...1, ...2) or could it still be adjusted?

**Reply by DIrgi** [reply]
That is, what we are right now discussing if it makes sense, are People really willing to use it* and devemux is thinking about if it's is technically manageable in general

*one should be clear: The blockings I forward to you come from another list others in the net are so gentle to maintain them.
When you want to have the in a more clearer way you have to separate them all out of the list, maintain your own one and put them into the right folders. I wonder if people will do that, we are all lazy (including me)

**Reply by devemux86** [reply]
> is this possible in Cruiser 5 with certain folder names (Forbidden Areas, ...1, ...2)

@umpfmpf groups in favorites, routes and tracks / areas can be freely renamed.

**Reply by umpfmpf** [reply]
There are lists that offer route closures for: motorcycle at night, motorcycle weekend, motorcycle always. 
In Cruiser 5 it is possible to create new folders, also with the option to use them as a route closure, but apparently only one folder created by the system works.

**Reply by umpfmpf** [reply]
![Screenshot_20240807_122040_Cruiser](https://github.com/user-attachments/assets/d04d96fe-f3f7-48c5-b27a-258b9d79210f)

**Reply by devemux86** [reply]
Cruiser automatically creates 1 group for favorites and 1 group for no-go areas.

You can create more groups and select if they will be for favorites or no-go areas.
Use the menu and select "New group".

**Reply by bios4** [reply]
> You can create more groups and select if they will be for favorites or no-go areas.
Use the menu and select "New group".

...which is currently available only in the "Favorites" section of My Collection!?

From my POV it would make sense to allow this also in the "Tracks/Areas" section, so that e.g. road closure tracks (e.g. the ones from DIrgi) could be added in the proper section.

What do you think?

**Reply by umpfmpf** [reply]
It's true: I can rename and create groups, but these are not considered prohibited routes.

**Reply by devemux86** [reply]
"Tracks / Areas" items and groups also have a "No-go area" switch in their menu.

Groups of tracks / areas can store items with different properties (colors, visible, no-go, ...).
While groups of favorites have the same icon, color, visible, no-go, etc.

**Reply by devemux86** [reply]
> It's true: I can rename and create groups, but these are not considered prohibited routes.

There are no "prohibited routes".

Only no-go circles and polygons:
- Circles in Favorites
- Polygons in Tracks / Areas

**Reply by bios4** [reply]
> "Tracks / Areas" items and groups also have a "No-go area" switch in their menu.

WHERE?

When selecting an existing group, where can I define it as a no-go area?

And: when creating a new group in "Tracks/Areas", I am not offered a "No-go area" checkbox like it is available when creating a new group in "Favorites" (this should be implemented in a similar way for the sake of consistency).

**Reply by bios4** [reply]
> WHERE?

The menu of a group offers:

- New group
- Rename
- Color
- Show on map
- Delete
- More...

And in "More":

- Import
- Export
- Share

**Reply by devemux86** [reply]
The routes and tracks / areas store their properties in the routes and tracks / areas (not in the groups).
They have just folders as groups. So import a polygon and set it as no-go area.

While favorites store their properties in their groups, there are no folders, just gpx files with the favorites.

**Reply by devemux86** [reply]
They do not need to be the same, they are completely different items.
Favorites vs Routes vs Tracks / Areas.

**Reply by bios4** [reply]
A track's menu offers:

- Rename
- Color
- Track diagram
- Show on map
- Center on map
- Calculate route
- Change group
- Delete
- More...

And in "More":

- Import
- Export
- Share
- Sort
- Mark all

So please once more: let us (me) know IN DETAIL where this is to be set up in "Tracks/Areas"!

**Reply by devemux86** [reply]
Only polygons can be set as no-go areas, not lines.

So import a polygon (i.e. closed track) and you will see the no-go switch.

**Reply by devemux86** [reply]
For example these GPX with closed track and route. Or any GeoJSON with polygons.
(areas have different icon from the tracks)

[nogo_Munich_MR-track.gpx.txt](https://github.com/user-attachments/files/16522790/nogo_Munich_MR-track.gpx.txt)
[nogo_Munich_MR-route.gpx.txt](https://github.com/user-attachments/files/16522792/nogo_Munich_MR-route.gpx.txt)

**Reply by bios4** [reply]
I see, thank you.

I got lost a bit in all of the current discussions, but is there a plan to extend this to (single line) tracks in the future?

**Reply by devemux86** [reply]
BRouter seems to support no-go lines.

However too many and complex no-go polygons and lines are not good for calculation speed.
So we should be careful.

**Reply by WolfgangZ42** [reply]
> Only polygons can be set as no-go areas, not lines.

Okay, that has a certain logic. But until now, no-go areas in the form of a track were also possible and in particular environments better than polygons. 
Sure the result for routing is the same in both versions:
<img width="200" src="https://github.com/user-attachments/assets/33457b57-1392-423c-ad58-819163435280"/> . . . <img width="200" src="https://github.com/user-attachments/assets/955f9af3-ab0c-432f-b4e5-613c744d66ca"/>

So - that's fine for me in principle. **But:** because "My collection" applies to all profiles, I will no longer be able to assign the no-go area only to the "Car" profile by default, but will have to make changes manually in "My collection" when changing profiles. This is a step backwards. We discussed that already more than once ...

Btw: Since I am a perfectionist (unfortunately?), I have created a geojson file in BRouter-web that contains the exact contour of the Munich Diesel Ban. This can be loaded in BRouter-web as a no-go area and works perfectly. 
<img width="250" src="https://github.com/user-attachments/assets/bfc6c49e-2dc9-4662-b90f-5138c3b0fd05"/> . . . <img width="250" src="https://github.com/user-attachments/assets/c3923751-9e08-4dde-804a-ebf7cc9ddddd"/> . . . <img width="250" src="https://github.com/user-attachments/assets/89a00363-c83d-44b7-92c3-631645e33471"/> 

I can also load it into Cruiser 5, "show on map" also works, but it is not recognized as a no-go area and cannot be activated. 

<img width="200" src="https://github.com/user-attachments/assets/0ac4354e-7581-4c45-a463-cceca150f9b0"/> . . . <img width="200" src="https://github.com/user-attachments/assets/a4d977e2-1596-4e02-8de1-b278dfcadbb1"/>

Here the geojson file: [no-go Munich DB.geojson.txt](https://github.com/user-attachments/files/16533053/no-go.Munich.DB.geojson.txt)

**Reply by Ciutnik** [reply]
@WolfgangZ42 No-go zones work well, but require a closed area. In my city, I drew a zone in which the first point exactly coincides with the last one. 

An open path has an icon different from a closed path (area). Only the one with a square in the icon can be used as a no-go.

<img src="https://github.com/user-attachments/assets/73b575a0-285d-4ae3-a1a9-92bf0bb43844" width="75%"/>

**Reply by Ciutnik** [reply]
Moreover, on the map you see a filled area, not just an outline:

<img src="https://github.com/user-attachments/assets/a7cc255a-e2b9-4ded-a6ec-01f19da2328a" width="75%"/>

**Reply by DIrgi** [reply]
If you identify the borders for no go areas or zones you can use overpass to support your extraction needs out of the data jungle. I did this for all so called crit‘air environment zones in France, worked really well. Now I can just switch them on to have them visible.

**Reply by WolfgangZ42** [reply]
@Ciutnik Could you please attach your file here - just to take a look and to identify the differences? Thanks!

**Reply by Ciutnik** [reply]
Of course. It's a simple gpx, but I guess it doesn't matter.

[nogo_SCT-Warszawa.gpx.txt](https://github.com/user-attachments/files/16533786/nogo_SCT-Warszawa.gpx.txt)

**Reply by devemux86** [reply]
@WolfgangZ42 

> But: because "My collection" applies to all profiles, I will no longer be able to assign the no-go area only to the "Car" profile by default, but will have to make changes manually in "My collection" when changing profiles. This is a step backwards. We discussed that already more than once ...

Maybe you missed the news, but now the visibility of the favorites, routes, tracks / areas can be customized per profile.
So enable them in one profile, disable them in the other profile. When they are enabled they are used for their purpose.
Overlays for display, no-go areas for routing, etc.

- https://github.com/devemux86/cruiser/discussions/597

**Reply by devemux86** [reply]
With GPX we need to check if the track / route is "closed", i.e. starts / ends at same point to import it as polygon.

GeoJSON specification contains all types: Point, LineString, Polygon, MultiPoint, MultiLineString and MultiPolygon.

See here for details and how each type is written: https://en.wikipedia.org/wiki/GeoJSON

---

@WolfgangZ42 

Your GeoJSON is not a polygon. It contains a "LineString" type, so all readers load it as line.

**Reply by WolfgangZ42** [reply]
> Maybe you missed the news

Sorry for complaining unnecessarily Emux - you are right, I didn't get that. Now tested with my "old" polygon of 2023 - works! 
I'll stick to this approach and give up the GeoJSON line - there is no real reason for "fine tuning" the polygon. 
Thanks for your patience.

**Reply by WolfgangZ42** [reply]
> It's a simple gpx, but I guess it doesn't matter.

@Ciutnik Sometimes simple is better than high-sophisticated ... my thoughts regarding GeoJSON went in the wrong direction. 
Your gpx file is quite identical with mine. 
Now that I have Poland implemented in my maps, I should plan a tour there. Ten years ago, I went to Masuria with my wife, but without a motorcycle - we have fond memories of that.

**Comment by 0709wiwiwi** [comment]
Up until now (v 4) the gpx track worked like a nogo even if not perfectly closed. Start and end position at a place without any intermediate roads the router could not get in anyway.  As a cyclist, go zones are more important than nogo ;-) Anyway I do not test the beta v 5 others have the honor to pull the chestnuts out of the fire ;-)

**Reply by devemux86** [reply]
@0709wiwiwi if you mean loading external BRouter no-go files,
this was handled by BRouter without us knowing what happens.

Also the [no-go file names](https://github.com/abrensch/brouter/blob/master/docs/users/android_advanced.md#using-nogo-areas) were a bit confusing with restrictions, so many users had errors.

As was already mentioned, we can add a no-go switch to tracks too, if there is real interest.
Just remember that complex no-go make route calculation slower, so don't complain later. :slightly_smiling_face: 

And polygons or tracks can be very complex, users do not understand it,
they just enable switches and expect everything to work with the same speed.

**Reply by Ciutnik** [reply]
> Just remember that complex no-go make route calculation slower, so don't complain later.

I see another problem here, which has already been mentioned here: the no-go path will block the passage **across** as well. And this may not be obvious to newcomers.

**Reply by 0709wiwiwi** [reply]
> the no-go path will block the passage **across** as well. And this may not be obvious to newcomers.

Agree.

**Reply by 0709wiwiwi** [reply]
Still, this. I checked with Wolfgang first.  For that middle ring in Munich, on the other hand, it is advantageous. The nogo is (must be) including that ring. A counter clock wise design.

**Reply by WolfgangZ42** [reply]
Yes Willy, it was perfectly the contour of "Mittlerer Ring" and it worked excellent. But right now I threw away all this stuff and only use my polygon which does the job as well.

**Comment by devemux86** [comment, marked as answer]
- Use no-go areas in route planning in https://github.com/devemux86/cruiser/discussions/641

For instructions please see [above](https://github.com/devemux86/cruiser/discussions/538#discussion-6671292).

**Comment by MotoUKRider** [comment]
I tried this feature on the latest beta, and noticed that creating/sizing requires a numeric input rather than a pinch/zoom-draw action.
Surely the latter is more user friendly as you can easily adjust its size by sight on map rather than guesswork?

**Reply by devemux86** [reply]
@MotoUKRider Probably, but it requires a lot of work to implement such a UI, at least for now. :slightly_smiling_face:
